subprojects/docs/src/docs/userguide/gradleWrapper.adoc

```diff
@@ -15,106 +15,204 @@
+[[sec:executing_wrapper]]
+=== Executing the build with the Wrapper
 
-If you’re wondering where the Gradle distributions are stored, you’ll find them in your user home directory under `$USER_HOME/.gradle/wrapper/dists`.
+It is recommended to always execute a build with the Wrapper to ensure a reliable, controlled and standardized execution of the build. Using the Wrapper looks almost exactly like running the build with a Gradle installation. Depending on the operation system you either run `gradlew` or `gradlew.bat` instead of the `gradle` command. The following console output demonstrate the use of the Wrapper on a Windows machine for a Java-based project.
 
-[[sec:wrapper_generation]]
-=== Adding the Wrapper to a project
+++++
+<sample xmlns:xi="http://www.w3.org/2001/XInclude" dir="userguide/wrapper" id="wrapperBatchFileExecution" title="Executing the build with the Wrapper batch file">
+    <output executable="gradlew.bat" args="build" />
+</sample>
+++++
 
-The Wrapper is something you _should_ check into version control. By distributing the Wrapper with your project, anyone can work with it without needing to install Gradle beforehand. Even better, users of the build are guaranteed to use the version of Gradle that the build was designed to work with. Of course, this is also great for http://en.wikipedia.org/wiki/Continuous_integration[continuous integration] servers (i.e. servers that regularly build your project) as it requires no configuration on the server.
+In case the Gradle distribution is not available on the machine, the Wrapper will download it and store in the local file system. Any subsequent build invocation is going to reuse the existing local distribution as long as the distribution URL in the Gradle properties doesn't change.
 
-You install the Wrapper into your project by running the `wrapper` task. (This task is always available, even if you don't add it to your build). To specify a Gradle version use `--gradle-version` on the command-line. By default, the Wrapper will use a `bin` distribution. This is the smallest Gradle distribution. Some tools, like Android Studio and Intellij IDEA, provide additional context information when used with the `all` distribution. You may select a different Gradle distribution type by using `--distribution-type`. You can also set the URL to download Gradle from directly via `--gradle-distribution-url`. If no version or distribution URL is specified, the Wrapper will be configured to use the gradle version the `wrapper` task is executed with. So if you run the `wrapper` task with Gradle 2.4, then the Wrapper configuration will default to version 2.4.
+[NOTE]
+====
+The Wrapper shell script and batch file reside in the root directory of a single or multi-project Gradle build. You will need to reference the correct path to those files in case you want to execute the build from a subproject directory e.g. `../../gradlew tasks`.
+====
+
+[[sec:adding_wrapper]]
+=== Adding a Wrapper to a project
```

Comment:
As a rule of thumb, we should keep subsection titles to under 25 characters or so. These will end up in the side navigation soon. We will want to adjust most all of the titles in this doc.